libsolidity/ast/Types.cpp

```diff
@@ -3453,6 +3453,16 @@ FunctionTypePointer FunctionType::asExternallyCallableFunction(bool _inLibrary,
 			parameterTypes.push_back(t);
 	}
 
+	TypePointers returnParameterTypes;
+	for (auto const& returnParamType: m_returnParameterTypes)
+	{
+		auto refType = dynamic_cast<ReferenceType const*>(returnParamType);
```

Comment:
This could all use `TypeProvider::withLocationIfReference` - the problem I'm having is that it sets "pointer" to false. I'm wondering if thet is the right behaviour...

Reply:
Ok, it turn out both are useful. Can you add a "isPoninter" parametre to `withLocationIfReference`?

Reply:
I did change `withLocationIfReference` with additional parameter.
Makes for loop easier. Not sure if it is the right behavior since it will set data location for all reference types to memory

Reply:
Oh you are right! We still should only change calldata to memory - storage has to stay storage.

Reply:
Maybe add another function in `TypeProvider::withLocationIfReference(_originalLocation, _targetLocation, ...)`. Or do it as it was before with cast here?

Reply:
What does ``isPointer`` actually mean for ``calldata``? And everything in ``memory`` is *always* a pointer, isn't it? So the ``isPointer`` thing is only a ``storage`` thing?

Reply:
Yes, see the documentation of `ReferenceType::isPointer`.